packages/core-sdk/src/resources/spg.ts

```diff
@@ -0,0 +1,65 @@
+import { PublicClient, getAddress, isAddress, zeroAddress } from "viem";
+
+import { SimpleWalletClient, SpgClient } from "../abi/generated";
+import {
+  CreateSPGNFTCollectionRequest,
+  CreateSPGNFTCollectionResponse,
+} from "../types/resources/spg";
+import { handleError } from "../utils/errors";
+
+export class SPGClient {
+  public spgClient: SpgClient;
+  private readonly rpcClient: PublicClient;
+  private readonly wallet: SimpleWalletClient;
+
+  constructor(rpcClient: PublicClient, wallet: SimpleWalletClient) {
+    this.rpcClient = rpcClient;
+    this.wallet = wallet;
+    this.spgClient = new SpgClient(rpcClient, wallet);
+  }
+
+  /**
+   * Creates a new SPG NFT Collection.
+   * @param request - The request object containing necessary data to create a SPG NFT Collection.
+   *   @param request.name - The name of the collection.
+   * 	 @param request.symbol - The symbol of the collection.
+   * 	 @param request.maxSupply - The maximum supply of the collection.
+   * 	 @param request.mintCost - The cost to mint a token.
+   * 	 @param request.mintToken - The token to mint.
+   * 	 @param request.owner - The owner of the collection.
+   *   @param request.txOptions - Optional transaction options.
+   * @returns A Promise that resolves to a CreateSPGNFTCollectionResponse containing the transaction hash and collection address.
+   * @emits CollectionCreated (nftContract);
+   */
+  public async createSPGNFTCollection<
+    TReq extends CreateSPGNFTCollectionRequest,
+    TRes = CreateSPGNFTCollectionResponse<TReq>,
+  >(request: TReq): Promise<TRes> {
+    try {
+      if (request.mintCost > 0n && !isAddress(request.mintToken || "")) {
+        throw new Error("Invalid mint token address, mint cost is greater than 0.");
+      }
+
+      const txHash = await this.spgClient.createCollection({
+        name: request.name,
+        symbol: request.symbol,
+        maxSupply: request.maxSupply,
+        mintCost: request.mintCost,
```

Comment:
I think mintCost can be optional too.

Reply:
Yes, and i think maxSupply should be optional too, in the case of open edition collections

Reply:
Resolved 7ee307b